pageserver/src/page_service.rs

```diff
@@ -1047,6 +1161,10 @@ impl PageServerHandler {
                         effective_request_lsn,
                         ctx,
                     }],
+                    // The executor grabs the batch when it becomes idle.
+                    // Hence, [`GetPageBatchBreakReason::ExecutorSteal`] is the
+                    // default reason for breaking the batch.
+                    batch_break_reason: GetPageBatchBreakReason::ExecutorSteal,
```

Comment:
Do we really want to hit a global counter for this common case?

I would be fine not having a counter for this common case

Reply:
We already update a global histogram in the common case, so I'd say it's fine.

It is a bit redundant since you can deduce the number of executor steals by looking at the total number of batches, but it's nice to have things labelled properly.

Reply:
Don't feel super strongly though. Can get rid of it if you do.